### PR TITLE
IIP Protocol defined in OpenAPI

### PR DESCRIPTION
**Pull request by VictorHarbo**

### CODE REVIEW

File: src/main/openapi/ds-image-openapi_v1.yaml [hunk 1]
```diff
@@ -16,395 +22,258 @@ servers:
...
-                $ref: '#/components/schemas/Error'
-        '500':
-          description: 'Internal Error'
-          content:
-            text/plain:
-              schema:
-                type: string
+          examples:
+            Rotate 180 degrees:
+              value: "180"
```
Comment (tokee) [inline src/main/openapi/ds-image-openapi_v1.yaml:177]: The first example is automatically added to the request through the OpenAPI GUI, so it should probably be `0` so users are not confused by seeing an upside-down image when the try the service.

File: src/main/openapi/ds-image-openapi_v1.yaml [hunk 2]
```diff
@@ -16,395 +22,258 @@ servers:
...
+          description: |- 
+            Color transformation to output space, c. Valid values are greyscale (GREY or GRAY) or to binary (BINARY). 
+            
+            Examples:
+            Convert to greyscale: COL=gray
+            
+            Convert to binary: COL=binary
+          schema: 
+              type: string
+              enum: ["gray", "binary"]
```
Comment (tokee) [inline src/main/openapi/ds-image-openapi_v1.yaml:252]: `grey` is also valid :-)

File: src/main/openapi/ds-image-openapi_v1.yaml [hunk 3]
```diff
@@ -16,395 +22,258 @@ servers:
 
 paths:
 
-  # TODO: Sample endpoint. Remove when building a concrete application
-  # This definition uses parameters: http://example.com/myservice/v1/hello?alternateHello=Boston
-  /hello:
+  /IIP:
```
Comment (tokee) [inline src/main/openapi/ds-image-openapi_v1.yaml:25]: I think this should be directly under wherever the service is deployed. I guess `/IIP:` should be just `/:`

**Review by tokee — CHANGES_REQUESTED** [review]
Overall looks good. Two syntactic problems and some OpenAPI idiosyncrasies.

File: src/main/openapi/ds-image-openapi_v1.yaml [hunk 4]
```diff
@@ -16,395 +22,258 @@ servers:
...
-           application/json:
-             schema:
-               $ref: '#/components/schemas/Book'
-           application/xml:
-             schema:
-               $ref: '#/components/schemas/Book'
-           application/x-www-form-urlencoded:
-             schema:
-               $ref: '#/components/schemas/Book'
+            enum: ["jpeg", "png"]
```
[the remark this answers is not in the document]
Comment (VictorHarbo) [reply src/main/openapi/ds-image-openapi_v1.yaml]: As i read the standard it does not seem to have a default value for CVT. Would it make sence if we chose one or should we leave it unspecified?

File: src/main/openapi/ds-image-openapi_v1.yaml (hunk 1 above)
Comment (VictorHarbo) [reply src/main/openapi/ds-image-openapi_v1.yaml:139]: I don't understand why Github does not mark this as outdated, The zero rotation example has been added.

File: src/main/openapi/ds-image-openapi_v1.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (tokee) [reply src/main/openapi/ds-image-openapi_v1.yaml]: Let's follow your specification-read and leave it.

File: src/main/openapi/ds-image-openapi_v1.yaml (hunk 1 above)
Comment (tokee) [reply src/main/openapi/ds-image-openapi_v1.yaml:139]: Looks like a bug in GitHub's system. The example is indeed added.

**Review by tokee — CHANGES_REQUESTED** [review]
You removed the endpoints `/monitor/ping`and `monitor/status`. These endpoints should be in "all" services, so please restore them in the OpenAPI YAML.

**Review by tokee — APPROVED** [review]

**Pull request merged**